Mathlib/AlgebraicTopology/Quasicategory/TwoTruncated.lean

```diff
@@ -125,28 +134,151 @@ lemma HomotopicR.trans [Quasicategory₂ X] {x y : X _⦋0⦌₂} {f g h : Edge
+-/
+lemma Edge.CompStruct.comp_unique {f f' : Edge x y} {g g' : Edge y z} {h h' : Edge x z}
+    (s : CompStruct f g h) (s' : CompStruct f' g' h')
+    (hf : HomotopicL f f') (hg : HomotopicL g g') : HomotopicL h h' := by
+  rcases hg.homotopicR with ⟨hg⟩
+  rcases hf with ⟨hf⟩
+  let ⟨s₁⟩ := Quasicategory₂.fill32 hf (idComp g') s'
+  let ⟨s₂⟩ := Quasicategory₂.fill31 (compId f) hg s₁
+  exact Quasicategory₂.fill31 s (compId g) s₂
+
+/--
+Given two consecutive edges `f`, `g`  in a 2-truncated quasicategory, nonconstructively choose:
+* an edge that is the diagonal of a 2-simplex with spine given by `f` and `g`, and
+* a 2-simplex witness this property.
+-/
+noncomputable
+def composeEdges (f : Edge x y) (g : Edge y z) :=
+  Nonempty.some (Quasicategory₂.fill21 f g)
+
+/--
+The edge `composeEdges f g` is the unique edge up to homotopy such that there is
+a 2-simplex with spine given by `f` and `g`
+-/
+lemma composeEdges_unique {f : Edge x y} {g : Edge y z} {h : Edge x z}
+    (s : CompStruct f g h) : HomotopicL h (composeEdges f g).1 :=
+  comp_unique s (composeEdges f g).2 .refl .refl
+
+/--
+Given two pairs of composable edges `f`, `g` and `f'`, `g'` such that `f` ≃ `f'` and
+`g` ≃ `g'`, their composites `h` and `h'` chosen by `composeEdges` are homotopic.
+-/
+lemma composeEdges_homotopic {f f' : Edge x y} {g g' : Edge y z}
+    (hf : HomotopicL f f') (hg : HomotopicL g g') :
+    HomotopicL (composeEdges f g).1 (composeEdges f' g').1 :=
+  comp_unique (composeEdges f g).2 (composeEdges f' g').2 hf hg
+
+/--
+The homotopy category of a 2-truncated quasicategory `A` has as objects the vertices of `A`
+-/
+structure HomotopyCategory₂ (A : Truncated 2) where
+  /-- An object of the homotopy category is a vertex of `A`. -/
+  pt : A _⦋0⦌₂
+
+/--
+Left homotopy is an equivalence relation on the edges of `A`.
+Remark: We could have equivalently chosen right homotopy, as shown by `homotopicL_iff_homotopicR`.
+-/
+instance instSetoidEdge (x y : A _⦋0⦌₂) : Setoid (Edge x y) where
+  r := HomotopicL
+  iseqv := ⟨fun _ ↦ HomotopicL.refl, HomotopicL.symm, HomotopicL.trans⟩
+
+/--
+The morphisms between two vertices `x`, `y` in `HomotopyCategory₂ A` are homotopy classes
+of edges between `x` and `y`.
+-/
+def HomotopyCategory₂.Hom (x y : HomotopyCategory₂ A) := Quotient (instSetoidEdge x.pt y.pt)
+
+/--
+Composition of morphisms in `HomotopyCategory₂ A` is given by lifting the edge
+chosen by `composeEdges`.
+-/
+noncomputable
+instance : CategoryStruct (HomotopyCategory₂ A) where
+  Hom x y := HomotopyCategory₂.Hom x y
+  id x := Quotient.mk' (Edge.id x.pt)
+  comp := Quotient.lift₂
+    (fun f g ↦ ⟦(composeEdges f g).1⟧)
+    (fun _ _ _ _ hf hg ↦ Quotient.sound (composeEdges_homotopic hf hg))
+
+/--
+Any edge in the 2-truncated simplicial set `A` defines a morphism in the homotopy category
+by taking its equivalence class.
+-/
+def homMk {x y : HomotopyCategory₂ A} (f : Edge x.pt y.pt) : x ⟶ y := ⟦f⟧
```

Comment:
It may be slightly better to make `x y : A _⦋0⦌₂` and assume `f : Edge x y`, so that we get a morphism `.mk x  ⟶ .mk y`. This makes it easier to transform an `Edge` into a morphism, because we no longer need to specify `x` and `y` as objects of the `HomotopyCategory₂`.
(This way of having a constructor `mk` for objects, and a constructor `homMk` for morphisms between `mk _` is used in other places in mathlib, like for the free groupoid of a category.)